SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialSpanIdStack.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/Connection/SpatialSpanIdStack.h"
+
+#include "Interop/Connection/SpatialEventTracer.h"
+
+namespace SpatialGDK
+{
+FSpatialSpanIdStack::FSpatialSpanIdStack(const SpatialEventTracer* InEventTracer)
+	: EventTracer(InEventTracer)
+{
+}
+
+void FSpatialSpanIdStack::SetEventTracer(const SpatialEventTracer* InEventTracer)
+{
+	EventTracer = InEventTracer;
+}
+
+void FSpatialSpanIdStack::Stack(const Trace_SpanId& SpanId)
+{
+	SpanIdStack.Add(SpanId);
+}
+
+void FSpatialSpanIdStack::Add(const Trace_SpanId& SpanId)
+{
+	const int32 Size = SpanIdStack.Num();
+	if (Size == 0)
+	{
+		Stack(SpanId);
+		return;
+	}
+
+	const Trace_SpanId& TopSpanId = SpanIdStack[Size - 1];
+	Trace_SpanId MergeCauses[2] = { SpanId, TopSpanId };
+	SpanIdStack[Size - 1] = EventTracer->CreateSpan(MergeCauses, 2).GetValue();
```

Comment:
It looks like when tracing properties, we will always call `Add` when adding a span, and looking at this function it will always merge them and replace the top item on the stack, so it will actually just stay at size 1.
If that is the case, we don't need to store those in `FSpatialSpanIdStack`, instead just storing a single span in the map that we will then merge when adding new spans.

Reply:
This also makes me think that perhaps having this class is unnecessary (unless we expect the implementation to change soon)